### PR TITLE
nsc-event-android-1-52-Event-DTO

### LINKED ISSUE #52: Create Event DTO
As a frontend developer I will create a data structure (DTO) for validation and passing events over the network.
DTO = data transfer object.

### PR DESCRIPTION
**Pull request by TroyFPV48**
Resolves #52

### CODE REVIEW

File: app/src/main/java/com/example/nsc_events/screen/AddEvent.kt [hunk 1]
```diff
@@ -18,24 +24,49 @@ import androidx.compose.material3.TopAppBar
...
 fun AddEventPage(navController: NavHostController) {
-    var eventName by remember { mutableStateOf("") }
+    var eventTitle by remember { mutableStateOf("") }
     var eventDescription by remember { mutableStateOf("") }
-    var eventDate by remember { mutableStateOf("") }
-
+    var eventCategory by remember { mutableStateOf("") }
+    var eventDate by remember { mutableStateOf(Calendar.getInstance().time) }
+    var eventStartDate by remember { mutableStateOf(Calendar.getInstance().time) }
+    var eventEndDate by remember { mutableStateOf(Calendar.getInstance().time) }
```
Comment (taylorpapke) [inline app/src/main/java/com/example/nsc_events/screen/AddEvent.kt:50]: I believe startDate and endDate should use String and be relabeled to startTime and endTime to match the name and type in our schema https://github.com/SeattleColleges/nsc-events-nestjs/blob/main/src/activity/schemas/activity.schema.ts

File: app/src/main/java/com/example/nsc_events/data/Event.kt [hunk 2]
```diff
@@ -2,10 +2,38 @@ package com.example.nsc_events.data
...
-    val eventName: String,
-    val description: String,
+    val eventTitle: String,
+    val eventDescription: String,
     /* TODO: change date variable to a Date data type instead of string */
-    val date: String
+    val eventCategory: String,
+    val date: Date,
+    val eventStartTime: Date,
+    val eventEndTime: Date,
```
Comment (taylorpapke) [inline app/src/main/java/com/example/nsc_events/data/Event.kt:19]: I believe eventStartTime and eventEndTime should be changed to String type to match our schema https://github.com/SeattleColleges/nsc-events-nestjs/blob/main/src/activity/schemas/activity.schema.ts

**Review by taylorpapke — CHANGES_REQUESTED** [review]
Overall looks great but after testing, I noticed that the back button to the login screen has a text field overlapping it which makes it hard to click:
Screenshot 2023-10-22 at 1 57 49 PM
I also noticed that some of the data types in nestjs are type String[] but we are using String here. @theGaryLarson is this fine for now or do you want us to start implementing the String[] right away?

**Comment by theGaryLarson** [comment]
> Overall looks great but after testing, I noticed that the back button to the login screen has a text field overlapping it which makes it hard to click: Screenshot 2023-10-22 at 1 57 49 PM I also noticed that some of the data types in nestjs are type String[] but we are using String here. @theGaryLarson is this fine for now or do you want us to start implementing the String[] right away?


The following attributes need to be String arrays:
* eventTags
* eventSpeakers

@TroyFPV48  please make these requested changes so we can approve, thanks.

**Comment by TroyFPV48** [comment]
sure I will fix the String elements again!

**Comment by taylorpapke** [comment]
@TroyFPV48 let's also not forget to resolve the conflicts on this so we can get it merged in.

**Comment by TroyFPV48** [comment]
> > > Overall looks great but after testing, I noticed that the back button to the login screen has a text field overlapping it which makes it hard to click:
> > 
> > 
> > Screenshot 2023-10-22 at 1 57 49 PM
> > I also noticed that some of the data types in nestjs are type String[] but we are using String here. @theGaryLarson is this fine for now or do you want us to start implementing the String[] right away?
> > We can probably trim some of these attributes of event. @Robel-003 what were the main things that they have asked tracked. I was thinking just the basics for now. Once everything is connected and functioning it will not be too difficult to come and add in additional attributes.
> > The following attributes need to be String arrays:
> > 
> > * eventTags
> > 
> > @TroyFPV48 please make these requested changes so we can approve, thanks.
> 
> In addition to the fields, they wanted a date, location, and an option to insert a flier or image, like an attach image button

Sorry, I wasn't cleared. Which fields specifically need attachable items?

**Comment by taylorpapke** [comment]
@Robel-003 what do you think? Can we waive the attachable items on this PR and get it on the next one since this is getting to be a fairly large merge?

**Review by TroyFPV48 — COMMENTED** [review]
I have fixed the conflicts, please review this PR.

**Comment by Robel-003** [comment]
> I have fixed the conflicts, please review this PR.

hey @TroyFPV48, it's still showing unresolved conflicts in the Event.kt and AddEvent.kt files

**Comment by taylorpapke** [comment]
@TroyFPV48 helpful doc on resolving merge conflicts: https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/addressing-merge-conflicts/resolving-a-merge-conflict-on-github

**Review by taylorpapke — APPROVED** [review]

**Review by Robel-003 — APPROVED** [review]
This looks good. I like the calendar options for date selections and I'm able to see the back arrow for the Login page

**Pull request merged**